### PR TITLE
forms 7.x | custom data providers | LRDOCS-7884

### PR DESCRIPTION
**Pull request by rbohl**
https://issues.liferay.com/browse/LRDOCS-7884

Hi Jim, you've seen this one already, but since then I've removed the "URL" field from the data provider's settings form, and instead hardcoded the url to the XML provider we're using as an example. We plan to instruct the reader to add the URL setting in the tutorial. Let me know what you think.

cc @carolmariaabb @natocesarrego

### CODE REVIEW

**Comment by jhinkey** [comment]
Started reviewing. My new PC is up and running. Thanks for your patience.

File: docs/dxp/latest/en/process-automation/forms/developer-guide/writing-a-custom-data-provider/resources/liferay-b4d8.zip/b4d8-impl/src/main/java/com/acme/b4d8/dynamic/data/mapping/data/provider/internal/B4D8DataProvider.java [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+import org.osgi.service.component.annotations.Reference;
+
+import org.w3c.dom.Document;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+
+import org.xml.sax.InputSource;
+
+@Component(
+	immediate = true, property = "ddm.data.provider.type=b4d8",
```
Comment (jhinkey) [inline docs/dxp/latest/en/process-automation/forms/developer-guide/writing-a-custom-data-provider/resources/liferay-b4d8.zip/b4d8-impl/src/main/java/com/acme/b4d8/dynamic/data/mapping/data/provider/internal/B4D8DataProvider.java:41]: @rbohl Can we tear out `immediate = true`? Does it work without it here and in the other class?

Comment (rbohl) [reply docs/dxp/latest/en/process-automation/forms/developer-guide/writing-a-custom-data-provider/resources/liferay-b4d8.zip/b4d8-impl/src/main/java/com/acme/b4d8/dynamic/data/mapping/data/provider/internal/B4D8DataProvider.java:41]: good question--i didn't test it without the property--slipped by me

Comment (jhinkey) [reply docs/dxp/latest/en/process-automation/forms/developer-guide/writing-a-custom-data-provider/resources/liferay-b4d8.zip/b4d8-impl/src/main/java/com/acme/b4d8/dynamic/data/mapping/data/provider/internal/B4D8DataProvider.java:41]: Could you test it? :-)

Comment (rbohl) [reply docs/dxp/latest/en/process-automation/forms/developer-guide/writing-a-custom-data-provider/resources/liferay-b4d8.zip/b4d8-impl/src/main/java/com/acme/b4d8/dynamic/data/mapping/data/provider/internal/B4D8DataProvider.java:41]: tested, works

Comment (jhinkey) [reply docs/dxp/latest/en/process-automation/forms/developer-guide/writing-a-custom-data-provider/resources/liferay-b4d8.zip/b4d8-impl/src/main/java/com/acme/b4d8/dynamic/data/mapping/data/provider/internal/B4D8DataProvider.java:41]: Thanks!

**Comment by rbohl** [comment]
Superseded by https://github.com/jhinkey/liferay-learn/pull/66

**Pull request closed without merging**